jdaviz/app.py

```diff
@@ -1020,13 +1027,10 @@ def get_subsets(self, subset_name=None, spectral_only=False,
                 else:
                     all_subsets[label] = subset_region
 
-        # can this be done at the top to avoid traversing all subsets if only
-        # one is requested?
-        all_subset_names = [subset.label for subset in dc.subset_groups]
-        if subset_name and subset_name in all_subset_names:
+        if subset_name and subset_name in all_subsets:
```

Comment:
Since you already filtered by subset_name above, why wouldn't subset_name be in all_subsets at this point? Is the second check necessary?